tensorboard/backend/application.py

```diff
@@ -338,34 +354,24 @@ def reload_multiplexer(multiplexer, path_to_run):
   tf.logging.info('TensorBoard done reloading. Load took %0.3f secs', duration)
 
 
-def start_reloading_multiplexer(multiplexer, path_to_run, load_interval):
-  """Starts a thread to automatically reload the given multiplexer.
+def continuously_reload_multiplexer(multiplexer, path_to_run, load_interval):
+  """Continuously reloads the given multiplexer after a given time interval.
 
-  The thread will reload the multiplexer by calling `ReloadMultiplexer` every
-  `load_interval` seconds, starting immediately.
+  Continuously reloads the multiplexer after `load_interval` seconds passes
+  since the last reload ended. Begins a reload immediately.
 
   Args:
-    multiplexer: The `EventMultiplexer` to add runs to and reload.
+    multiplexer: The `EventMultiplexer` to add runs to and continuously reload.
     path_to_run: A dict mapping from paths to run names, where `None` as the run
       name is interpreted as a run name equal to the path.
     load_interval: How many seconds to wait after one load before starting the
       next load.
-
-  Returns:
-    A started `threading.Thread` that reloads the multiplexer.
   """
-
   # We don't call multiplexer.Reload() here because that would make
   # AddRunsFromDirectory block until the runs have all loaded.
-  def _reload_forever():
-    while True:
-      reload_multiplexer(multiplexer, path_to_run)
-      time.sleep(load_interval)
-
-  thread = threading.Thread(target=_reload_forever, name='Reloader')
-  thread.daemon = True
-  thread.start()
-  return thread
+  while True:
+    reload_multiplexer(multiplexer, path_to_run)
+    time.sleep(load_interval)
```

Comment:
Hmm, it seems a bit simpler to me to just leave `start_reloading_multiplexer` exactly as it was, except call it unconditionally (regardless of reload_interval) and then change the body of _reload_forever() here to just do:

```
def _reload():
  while True:
    reload_multiplexer(multiplexer, path_to_run)
    if load_interval > 0:
      time.sleep(load_interval)
      continue
    return
```

Reply:
Hm, actually I forgot load_interval < 0 means never reload.  So that logic should probably still be in an "if load_interval >= 0: start_reloading_multiplexer()" condition up above (it could also be done by checking inside _reload() but then we'd always start a thread even if it had nothing to do).